casper/src/main/scala/coop/rchain/casper/merging/BlockIndex.scala

```diff
@@ -0,0 +1,70 @@
+package coop.rchain.casper.merging
+
+import cats.effect.Concurrent
+import cats.syntax.all._
+import coop.rchain.casper.protocol.ProcessedDeploy
+import coop.rchain.casper.util.EventConverter
+import coop.rchain.models.BlockHash.BlockHash
+import coop.rchain.models.block.StateHash.StateHash
+import coop.rchain.rspace.hashing.Blake2b256Hash
+import coop.rchain.rspace.history.HistoryRepository
+import coop.rchain.rspace.merger._
+import coop.rchain.rspace.syntax._
+import coop.rchain.rspace.trace.Produce
+
+import scala.collection.concurrent.TrieMap
+
+final case class BlockIndex(blockHash: BlockHash, deployChains: Vector[DeployChainIndex])
+
+object BlockIndex {
+
+  // TODO make proper storage for block indices
+  val cache = TrieMap.empty[BlockHash, BlockIndex]
```

Comment:
This is one task for future work - to persist index, and remove it once block is finalized.

Reply:
Might be this task is dependent on https://github.com/rchain/rchain/pull/3388, as in current finalizer only blocks that is finalized is the next LFB